clients/drcachesim/tracer/instr_counter.cpp

```diff
@@ -142,7 +143,13 @@ hit_instr_count_threshold(app_pc next_pc)
     instr_count = 0;
 #endif
     DR_ASSERT(tracing_mode.load(std::memory_order_acquire) == BBDUP_MODE_COUNT);
-    tracing_mode.store(BBDUP_MODE_TRACE, std::memory_order_release);
+
+    if (op_L0_filter_until_instrs.get_value()) {
```

Comment:
Yes, since we would simulate the samples independently we want the L0 filter at every sample. I will document this. Currently I have a single file with the filtered and regular traces separated by the endpoint marker which was introduced recently.

Reply:
What is the filetype of the file?  Looks like our offline post-processed half-filtered file is marked with a filter type even though half is unfiltered (separated by the filter endpoint marker) -- is it the same here where the type is filtered and that's how every window starts?

Reply:
We were just discussing this separately: we all agree we should have a new file type as anyone looking at the trace will want to know up front what to expect.

Reply:
Should we add a new offline file type (e.g. OFFLINE_FILE_TYPE_UNFILTERED) and then we could set both this bit along with the filter bits for the mixed files?

Reply:
The complexity with an unfiltered type is that existing unfiltered traces do not have it set so you'd have to bump the version and add legacy-version logic support.

Reply:
Should we do this as a separate PR?

Reply:
We can update the file type in a separate PR.